koku/masu/processor/orchestrator.py

```diff
@@ -542,3 +543,32 @@ def remove_expired_report_data(self, simulate=False):
             )
             async_results.append({"customer": account.get("customer_name"), "async_id": str(async_result)})
         return async_results
+
+    def remove_expired_trino_partitions(self, simulate=False):
+        """
+        Removes expired trino partitions for each account.
+        """
+        async_results = []
+        schemas = set(
+            Provider.objects.filter(type=Provider.PROVIDER_OCP)
+            .values_list("customer__schema_name", flat=True)
+            .distinct()
+        )
+        # <QuerySet ['org1234567', 'org1234567', 'org1234567', 'org1234567']>
+        # distinct is not removing duplicates from this list, so using a set to reduce instead
```

Comment:
The reason this query is not working correctly is because Django is ordering by the api_provider.name which adds an extra select statement:
```
SELECT DISTINCT "api_customer"."schema_name", "api_provider"."name" FROM "api_provider" LEFT OUTER JOIN "api_customer" ON ("api_provider"."customer_id" = "api_customer"."id") WHERE "api_provider"."type" = 'OCP' ORDER BY "api_provider"."name" ASC LIMIT 21; args=('OCP',)
```

That ordering is coming from [here](https://github.com/project-koku/koku/blob/main/koku/api/provider/models.py#L108).

To get around this, you can add an `order_by()` to your query, which basically tells us to order by nothing:
```
            Provider.objects.order_by().filter(type=Provider.PROVIDER_OCP)
            .values_list("customer__schema_name", flat=True)
            .distinct()

```
which results in the follow sql and result:
```
>>> Provider.objects.order_by().filter(type=Provider.PROVIDER_OCP).values_list("customer__schema_name", flat=True).distinct()

DEBUG:django.db.backends:(0.040) SELECT DISTINCT "api_customer"."schema_name" FROM "api_provider" LEFT OUTER JOIN "api_customer" ON ("api_provider"."customer_id" = "api_customer"."id") WHERE "api_provider"."type" = 'OCP' LIMIT 21; args=('OCP',); alias=default
<QuerySet ['org1234567']>
>>> 
```

Reply:
Awesome! Thanks for digging into this one for me Michael. 🙇